object_goslice_test.go

```diff
@@ -271,3 +271,138 @@ func TestGoSliceToString(t *testing.T) {
+			threshold: 100,
+			val: &objectGoSlice{
+				baseObject: baseObject{
+					val: &Object{runtime: vm},
+				},
+				data: &[]interface{}{
+					valueInt(99),
+					valueInt(99),
+					valueInt(99),
+					valueInt(99),
+				},
+			},
+			// overhead + values
+			expectedMem: SizeEmptyStruct + SizeInt*4,
+			// overhead + values
+			expectedNewMem: SizeEmptyStruct + SizeInt*4,
+			errExpected:    nil,
+		},
+		{
+			name:      "should account for each value given slice with a nil value",
+			threshold: 100,
+			val: &objectGoSlice{
+				baseObject: baseObject{
+					val: &Object{runtime: vm},
+				},
+				data: &[]interface{}{nil},
+			},
+			// overhead + null
+			expectedMem: SizeEmptyStruct + SizeEmptyStruct,
+			// overhead + null
+			expectedNewMem: SizeEmptyStruct + SizeEmptyStruct,
+			errExpected:    nil,
+		},
+		{
+			name:      "should account slice length over threshold",
+			threshold: 20,
+			val: &objectGoSlice{
+				baseObject: baseObject{
+					val: &Object{runtime: vm},
+				},
+				data: &[]interface{}{
+					valueInt(99),
+					valueInt(99),
+					valueInt(99),
+					valueInt(99),
+				},
+			},
+			// overhead + values
+			expectedMem: SizeEmptyStruct + SizeInt*4,
+			// overhead + values
+			expectedNewMem: SizeEmptyStruct + SizeInt*4,
+			errExpected:    nil,
+		},
+		{
+			name:      "should account for nested slices",
+			threshold: 100,
+			val: &objectGoSlice{
+				baseObject: baseObject{
+					val: &Object{runtime: vm},
+				},
+				data: &[]interface{}{
+					[]interface{}{
+						valueInt(99),
+						valueInt(99),
+						valueInt(99),
+						valueInt(99),
+					},
+				},
+			},
+			// overhead + (value + len("length") + "length".value + prototype + ints)
+			expectedMem: SizeEmptyStruct + (SizeEmptyStruct + 6 + SizeEmptyStruct + (SizeEmptyStruct + SizeEmptyStruct) + SizeNumber*4),
+			// overhead + (value + len("length") with string overhead + "length".value + prototype + ints)
+			expectedNewMem: SizeEmptyStruct + (SizeEmptyStruct + (6 + SizeString) + SizeEmptyStruct + (SizeEmptyStruct + SizeEmptyStruct) + SizeNumber*4),
+			errExpected:    nil,
+		},
+		{
+			name:      "should account for reflect object",
+			threshold: 100,
+			val: &objectGoSlice{
+				baseObject: baseObject{
+					val: &Object{runtime: vm},
+				},
+				data: &[]interface{}{
+					&objectGoSlice{
+						baseObject: baseObject{
+							val: &Object{runtime: vm},
+						},
+						data: &[]interface{}{
+							valueInt(99),
+							valueInt(99),
+							valueInt(99),
+							valueInt(99),
+						},
+					},
+				},
+			},
+			// overhead + nested reflect object
+			expectedMem: SizeEmptyStruct + SizeEmptyStruct,
+			// overhead + nested reflect object
+			expectedNewMem: SizeEmptyStruct + SizeEmptyStruct,
+			errExpected:    nil,
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			total, newTotal, err := tc.val.MemUsage(NewMemUsageContext(vm, 100, 100, 100, tc.threshold, nil))
```

Comment:
[minor] IF you feel adventurous and want to add estimates to the `objectGoSlice` mem usage functions then you would use the threshold on the array and not on the object props like you did here. Otherwise you can just remove the threshold field in the test struct altogether and hardcode it. 

```suggestion
			total, newTotal, err := tc.val.MemUsage(NewMemUsageContext(vm, 100, 100, tc.threshold, 100, nil))
```